### PR TITLE
Implement factor collection for additions

### PR DESCRIPTION
**Pull request by jirikuncar**
* Adds tests for the most common use-cases.

### CODE REVIEW

File: sympy/physics/units/quantities.py [hunk 1]
```diff
@@ -117,9 +116,15 @@ def _collect_factor_and_dimension(expr):
             factor, dim = Quantity._collect_factor_and_dimension(expr.base)
             return factor ** expr.exp, dim ** expr.exp
         elif isinstance(expr, Add):
-            raise NotImplementedError
+            factor, dim = Quantity._collect_factor_and_dimension(expr.args[0])
```
Comment (asmeurer) [inline sympy/physics/units/quantities.py]: Are you doing `args[0]` to pull off constant factors? If so, you can use `as_coeff_add()`, which gives you `0` for the coeff if there is no rational/float constant factor.

Comment (jirikuncar) [inline sympy/physics/units/quantities.py]: No, I am using it because `reduce` would be unreadable.

Comment (jirikuncar) [inline sympy/physics/units/quantities.py]: @asmeurer I have also followed the example in https://github.com/sympy/sympy/pull/12936/files#diff-6790d0f6aee4060f95bd1eeca4dd74d4R110

**Comment by schymans** [comment]
This is great! Would it be possible to add support for `Abs()`? Currently, it is not evaluated:
```
from sympy.physics.units import Quantity, Dimension, length, time, meter, second
from sympy import Abs
v_w1 = Quantity('v_w1', length/time, 1.5*meter/second)
v_w2 = Quantity('v_w2', length/time, 2*meter/second)
expr = Abs(v_w1/2 - v_w2)
Quantity._collect_factor_and_dimension(expr)
```
returns
```
(Abs(v_w1/2 - v_w2), 1)
```
It should return
```
(1.25000000000000, Dimension(length/time))
```
Another problem:
```
expr = v_w1 - Abs(v_w1/2 - v_w2)
```
raises:
```
ValueError: summation of quantities of incompatible dimensions
```
This should not happen, as the above expression is perfectly valid. And this one gives an incorrect result:
```
expr = v_w1 - Abs(v_w1)
Quantity._collect_factor_and_dimension(expr)
```
returns:
```
(0, 1)
```

**Comment by asmeurer** [comment]
Abs should be easy to add. You just need to implement `_eval_Abs`.

**Pull request merged**